server/repository/src/db_diesel/item.rs

```diff
@@ -239,10 +247,36 @@ fn create_filtered_query(store_id: String, filter: Option<ItemFilter>) -> BoxedI
             .filter(store_dsl::id.eq(store_id.clone()))
             .into_boxed();
 
-        query = match is_visible {
-            Some(true) => query.filter(item_dsl::id.eq_any(visible_item_ids)),
-            Some(false) => query.filter(item_dsl::id.ne_all(visible_item_ids)),
-            None => query,
+        let item_ids_with_stock_on_hand = item_link_dsl::item_link
+            .select(item_link_dsl::item_id)
+            .inner_join(stock_on_hand_dsl::stock_on_hand)
+            .filter(stock_on_hand_dsl::available_stock_on_hand.gt(0.0))
+            .group_by(item_link_dsl::item_id)
+            .into_boxed();
+
+        query = match (is_visible, has_stock_on_hand) {
```

Comment:
bit of a nit pick - some weird cases aren't covered.
If I want `is_visible = Some(true) && has_stock_on_hand = Some(false)` then I will potentially get items which aren't visible and do have stock on hand.
Similarly the 'false' 'false' case is giving me items that aren't visible but do have stock